### PR TITLE
Dave b1992 code review

### PR DESCRIPTION
**Pull request by PerryBaran**

### CODE REVIEW

File: src/kata3.numberToReversedDigits.js [hunk 1]
```diff
@@ -1,3 +1,7 @@
-const numberToReversedDigits = (number) => {};
+const numberToReversedDigits = (number) => {
+ const reversedDigits = String(number).split('').reverse();
+ const result = reversedDigits.map((digitString)=> Number(digitString));
+ return result
```
Comment (PerryBaran) [inline src/kata3.numberToReversedDigits.js:4]: no need to save `result` to a variable before returning it, can simplify do `return reversedDigits.map((digitString)=> Number(digitString));`

File: src/kata2.booleanToWord.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
-const booleanToWord = (boolean) => {};
-
+const booleanToWord = (boolean) => {
+  return boolean ? 'Yes' : 'No';
```
Comment (PerryBaran) [inline src/kata2.booleanToWord.js:2]: good use of ternary operator

File: src/kata4.humanCatDogYears.js [hunk 3]
```diff
@@ -1,3 +1,21 @@
-const humanCatDogYears = (number) => {};
+const humanCatDogYears = (number) => {
+   let humanYears = number;
```
Comment (PerryBaran) [inline src/kata4.humanCatDogYears.js:2]: no need to assign humanYears to number, if you want to rename the variable just rename the parameter

File: src/kata4.humanCatDogYears.js [hunk 4]
```diff
@@ -1,3 +1,21 @@
...
+           catYears += 15
+           dogYears += 15;
+       }else if(i === 2){
+               catYears += 9
+               dogYears += 9
+           }else{
+                   catYears += 4
+                   dogYears += 5
+               }
+           }
```
Comment (PerryBaran) [inline src/kata4.humanCatDogYears.js:16]: inconsistent indentation, all if/else if/else statements should be on the same level of indentation.

File: src/kata4.humanCatDogYears.js [hunk 5]
```diff
@@ -1,3 +1,21 @@
-const humanCatDogYears = (number) => {};
+const humanCatDogYears = (number) => {
+   let humanYears = number;
+   let catYears = 0;
+   let dogYears = 0;
+   for (let i = 0; i < humanYears; i++){
+       if (i === 1){
```
Comment (PerryBaran) [inline src/kata4.humanCatDogYears.js:6]: should be a space between the if statement and the curly bracket

File: src/kata4.humanCatDogYears.js [hunk 6]
```diff
@@ -1,3 +1,21 @@
...
+           catYears += 15
+           dogYears += 15;
+       }else if(i === 2){
```
Comment (PerryBaran) [inline src/kata4.humanCatDogYears.js:9]: should be a space between the closing curly bracket and else if. Should also be a space between else if and the if statement and a space between that and the opening curly bracket e.g. `} else if (i === 2) {`

File: src/kata5.reachDestination.js [hunk 7]
```diff
@@ -1,3 +1,6 @@
-const reachDestination = (distance, speed) => {};
+const reachDestination = (distance, speed) => {
+    result = Math.round(distance/speed * 2) / 2;
+    return`I should be there in ${result} hours.`
```
Comment (PerryBaran) [inline src/kata5.reachDestination.js:3]: minor nitpick, but if `result = 1` you return the string `"I should be there in 1 hours."` instead of `"I should be there in 1 hour."`

File: test/kata1.fizzBuzz.test.js [hunk 8]
```diff
@@ -1,11 +1,20 @@
 const { fizzBuzz } = require("../src");
 
 describe("fizzBuzz", () => {
-  test("returns Fizz when passed a multiple of 3", () => {});
+  it("returns Fizz when passed a multiple of 3", () => {expect(fizzBuzz(3)).toBe('Fizz')
+  });
```
Comment (PerryBaran) [inline test/kata1.fizzBuzz.test.js:5]: the expect statement should be on a new line

File: test/kata1.fizzBuzz.test.js [hunk 9]
```diff
@@ -1,11 +1,20 @@
...
 
-  test("returns Buzz when passed a multiple of 5", () => {});
+  it("returns Buzz when passed a multiple of 5", () => {expect(fizzBuzz(5)).toBe('Buzz')
+  });
 
-  test("returns FizzBuzz when passed a multiple 3 and 5", () => {});
+  it("returns FizzBuzz when passed a multiple 3 and 5", () => {
+    expect (fizzBuzz (15)) .toBe( 'FizzBuzz')
```
Comment (PerryBaran) [inline test/kata1.fizzBuzz.test.js:11]: there should be no spaces on this line, and final semi-colon e.g. `expect(fizzBuzz(15)).toBe('FizzBuzz');`

File: test/kata1.fizzBuzz.test.js [hunk 10]
```diff
@@ -1,11 +1,20 @@
...
+    expect (fizzBuzz (30)) . toBe( 'FizzBuzz')
+    expect (fizzBuzz (90) ).toBe('FizzBuzz')
+  });
 
-  test("returns the number when it isn't a multiple of 3 or 5", () => {});
+  it("returns the number when it isn't a multiple of 3 or 5", () => {
+    expect (fizzBuzz(4)).toBe(4)
+    expect (fizzBuzz(11)).toBe(11)
+  });
```
Comment (PerryBaran) [inline test/kata1.fizzBuzz.test.js:19]: you should be using only one type of quotes, i.e. not mixing double and single quotation marks in a project

File: test/kata2.booleanToWord.test.js [hunk 11]
```diff
@@ -1,5 +1,10 @@
...
 
 describe("booleanToWord", () => {
-  // how do we create specs again???
-});
+  it('returns Yes when passed true', () => {
+    expect(booleanToWord(true)).toBe('Yes')
+  })
+  it('returns No when passed false', () =>{
+    expect(booleanToWord(false)).toBe('No')
+  })
```
Comment (PerryBaran) [inline test/kata2.booleanToWord.test.js:9]: should be an empty line between tests (i.e. `it(...)`) to make it more readable

File: test/kata4.humanCatDogYears.test.js [hunk 12]
```diff
@@ -1,3 +1,8 @@
 const { humanCatDogYears } = require("../src");
 
-// Look Ma, no handlebars!!!
+describe ('humanCatDogYears', () => {
+    it("returns array of human, cat and  dog years when passed human years", () => {expect(humanCatDogYears(10)).toEqual([10, 56, 64])
```
Comment (PerryBaran) [inline test/kata4.humanCatDogYears.test.js:4]: the expect statement should be on another line. I would also suggest adding test cases for edge cases, i.e. when human years = 1 or 2.

File: src/kata4.humanCatDogYears.js (hunk 6 above)
Comment (PerryBaran) [inline src/kata4.humanCatDogYears.js:9]: remember that you're starting counting from 0, if `humanYears = 1 or 2 `you get the wrong answer. I would suggest writing more tests to check for these edge cases

File: test/kata7.getEmployerRole.test.js [hunk 13]
```diff
@@ -1,5 +1,26 @@
...
+      {
+        name: "Jenny",
+
+        role: "Sales Associate",
+      },
+      {
+        name: "Javid",
+
+        role: "Human Recommended Reading Assistant",
+      },
```
Comment (PerryBaran) [inline test/kata7.getEmployerRole.test.js:20]: should remove the empty line between` name` and `role` in the objects

**Review by PerryBaran — COMMENTED** [review]
Mostly good, all tests pass. I've left suggestions throughout for improvements, mostly formatting issues.

Good stuff:
- good naming of variables
- all test cases pass

Improvement suggestions:
- formatting. If you've got red squiggly lines in your code editor, make sure to try and fix them. if you hover your mouse over them it will tell you what's causing that issue
- inconsistent use of double and single quotes
- more tests to test for edge cases
